### PR TITLE
Make Bandit the default web server for new Phoenix apps

### PR DESCRIPTION
**Pull request by mtrudel**
This PR:

* Makes Bandit the default web server for newly generated Phoenix apps
* Updates the Phoenix docs to reflect both Bandit and Cowboy as first-class supported adapters
* Keeps the default adapter as Cowboy if not explicitly specified for backwards compatibility
* Since Phoenix still supports (and runs CI) back to 1.11, Phoenix's internal tests are NOT updated to use Bandit here. There's a separate PR coming to add this, which will gate on Phoenix bumping to 1.12+

### CODE REVIEW

**Comment by mtrudel** [comment]
This is working and ready. I *think* I may have discovered a corner case in Thousand Island shutdown handling that came up in the long poll integration tests, but there's nothing to change here either way (in the meantime, those tests are a bit noisier and a lot slower than they ought to be).

File: lib/phoenix/endpoint/supervisor.ex [hunk 1]
```diff
@@ -202,6 +202,10 @@ defmodule Phoenix.Endpoint.Supervisor do
       render_errors: [view: render_errors(module), accepts: ~w(html), layout: false],
 
       # Runtime config
+
+      # Even though Bandit is the default in apps generated via the installer,
+      # we continue to use Cowboy as the default if not explicitly specified for
+      # backwards compatibility. TODO: Change this to default to Bandit in 2.0
```
Comment (mtrudel) [inline lib/phoenix/endpoint/supervisor.ex:208]: nb

**Review by josevalim — APPROVED** [review]
If integration tests pass, this is good to go IMO!

**Comment by mtrudel** [comment]
Integration test failing on Plug not being loaded now that we've dropped plug_cowboy. There's likely some loading issue somewhere; I'll figure it out

**Comment by mtrudel** [comment]
Integration issue is upstream at https://github.com/swoosh/swoosh/pull/857. Once this is fixed / released I'll update the dep here

**Comment by mtrudel** [comment]
Will ya look at that, all green and ready to go.

**Comment by krns** [comment]
We are using https in our local development setup and configured it like described in https://hexdocs.pm/phoenix/using_ssl.html#ssl-in-development. Today I tried Bandit but it crashes:

```
my-app-1       | [notice] Application my_app exited: MyApp.Application.start(:normal, []) returned an error: shutdown: failed to start child: MyAppWeb.Endpoint
my-app-1       |     ** (EXIT) shutdown: failed to start child: {MyAppWeb.Endpoint, :https}
my-app-1       |         ** (EXIT) an exception was raised:
my-app-1       |             ** (RuntimeError) Plug.SSL.configure/1 encountered error: the :otp_app option is required when setting relative SSL certfiles
my-app-1       |                 (bandit 1.1.3) lib/bandit.ex:262: Bandit.start_link/1
my-app-1       |                 (stdlib 5.2) supervisor.erl:420: :supervisor.do_start_child_i/3
my-app-1       |                 (stdlib 5.2) supervisor.erl:406: :supervisor.do_start_child/2
my-app-1       |                 (stdlib 5.2) supervisor.erl:390: anonymous fn/3 in :supervisor.start_children/2
my-app-1       |                 (stdlib 5.2) supervisor.erl:1258: :supervisor.children_map/4
my-app-1       |                 (stdlib 5.2) supervisor.erl:350: :supervisor.init_children/2
my-app-1       |                 (stdlib 5.2) gen_server.erl:980: :gen_server.init_it/2
my-app-1       |                 (stdlib 5.2) gen_server.erl:935: :gen_server.init_it/6
my-app-1       |                 (stdlib 5.2) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
```

Maybe there are some things that are missing on our side, but I think these should be documented.

**Comment by mtrudel** [comment]
@krns yep - that's a wrinkle that you have to fix manually when using raw Bandit. In the case of Phoenix, we know for certain that we have an `otp_app` value so it's safe / preferable to pull it in automatically. I'm adding a change to `Bandit.PhoenixAdapter` to do just that. should be up in an hour or so

**Comment by krns** [comment]
Bandit 1.2.0 works, thank you so much @mtrudel!

File: installer/lib/phx_new/generator.ex [hunk 2]
```diff
@@ -304,8 +304,7 @@ defmodule Phx.New.Generator do
   end
 
   defp get_web_adapter("cowboy"), do: {:plug_cowboy, "~> 2.7", Phoenix.Endpoint.Cowboy2Adapter}
-  # TODO bump bandit to 1.0 when it's released
-  defp get_web_adapter("bandit"), do: {:bandit, ">= 0.0.0", Bandit.PhoenixAdapter}
+  defp get_web_adapter("bandit"), do: {:bandit, "~> 1.0", Bandit.PhoenixAdapter}
```
Comment (chrismccord) [inline installer/lib/phx_new/generator.ex:307]:
```suggestion
  defp get_web_adapter("bandit"), do: {:bandit, "~> 1.2", Bandit.PhoenixAdapter}
```

**Comment by mtrudel** [comment]
* [x] https://youtu.be/ZLjWyanLHuk?t=1636

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Comment by mtrudel** [comment]
* [x] https://youtu.be/ZLjWyanLHuk?t=1634
